### PR TITLE
adds return to API endpoints after server errors

### PR DESCRIPTION
**Pull request by cmiller01**
some API calls were missing a `return` so they would keep going after encountering an error

### CODE REVIEW

File: api-service/api/main.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ func getAllMenuItems(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
 		fmt.Fprintf(w, "Failed to read menu items")
 		w.WriteHeader(http.StatusInternalServerError)
+		return
 	}
```
Comment (TylerAckerson) [inline api-service/api/main.go:40]: should this `w.WriteHeader(http.StatusOK)`?

Comment (cmiller01) [reply api-service/api/main.go:40]: I think the 500 status code seems reasonable, if we can't get the menu items (for some reason) we throw an error and stop

Comment (TylerAckerson) [reply api-service/api/main.go:40]: Yes, agree on the 500. But I mean on the non-err response right above this
```
json.NewEncoder(w).Encode(res.MenuItemsList)
```

Comment (cmiller01) [reply api-service/api/main.go:40]: ah, gotcha, no need, on the first write to `http.ResponseWriter` it'll set the header to "OK"

https://pkg.go.dev/net/http#ResponseWriter
> // If WriteHeader has not yet been called, Write calls
// WriteHeader(http.StatusOK) before writing the data.

**Review by TylerAckerson — APPROVED** [review]

**Pull request merged**